rest-client/src/main/java/io/apicurio/registry/client/request/RequestHandler.java

```diff
@@ -59,5 +64,21 @@ private void handleError(Throwable e) {
             }
             logger.log(Level.SEVERE, "Error getting call result", e);
         }
+
+
+        private static void checkIfDeprecated(okhttp3.Headers headers) {
+            String isDeprecated = headers.get(Headers.DEPRECATED);
+            if (isDeprecated != null) {
+                String id = headers.get(Headers.ARTIFACT_ID);
+                String version = headers.get(Headers.VERSION);
+                logIfDeprecated(() -> ArtifactState.DEPRECATED, id, version);
+            }
+        }
+
+        private static void logIfDeprecated(Supplier<ArtifactState> stateSupplier, String artifactId, Object version) {
+            if (stateSupplier.get() == ArtifactState.DEPRECATED) {
+                logger.warning(String.format("Artifact %s [%s] is deprecated", artifactId, version));
+            }
+        }
```

Comment:
The `stateSupplier` stuff is not really needed here, right?  This can just be a `logDeprecated` method like this:

```
private static void logDeprecated(String artifactId, Object version) {
    logger.warning(String.format("Artifact %s [%s] is deprecated", artifactId, version));
}
```

Reply:
Unless this method is intended to be used in other ways.